pkg/operator/rules.go

```diff
@@ -126,9 +127,80 @@ func (r *rulesReconciler) Reconcile(ctx context.Context, req reconcile.Request)
 	if err := r.ensureRuleConfigs(ctx, projectID, location, cluster); err != nil {
 		return reconcile.Result{}, fmt.Errorf("ensure rule configmaps: %w", err)
 	}
+
+	if err := r.scaleRuleConsumers(ctx); err != nil {
+		return reconcile.Result{}, fmt.Errorf("scale rule consumers: %w", err)
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *rulesReconciler) scaleRuleConsumers(ctx context.Context) error {
+	var desiredReplicas int32
+
+	var hasAnyRules bool
+	for _, check := range []RuleCheck{r.hasRules, r.hasClusterRules, r.hasGlobalRules} {
+		hasRules, err := check(ctx)
+		if err != nil {
+			return err
+		}
+		if hasRules {
+			hasAnyRules = true
+			break
+		}
+	}
+	if hasAnyRules {
+		desiredReplicas = 1
+	}
+
+	var alertManagerStatefulSet appsv1.StatefulSet
+	if err := r.client.Get(ctx, client.ObjectKey{Namespace: r.opts.OperatorNamespace, Name: "alertmanager"}, &alertManagerStatefulSet); err != nil {
+		return err
+	}
+	var ruleEvaluatorDeployment appsv1.Deployment
```

Comment:
add a check for the edge case if the rule-evaluator doesn't exist. Similar to above comment.